cartoframes/context.py

```diff
@@ -473,6 +473,64 @@ def sync(self, dataframe, table_name):
         """
         pass
 
+    def fetch(self, query, decode_geom=False):
```

Comment:
A comment about `decode_geom`: I am not sure if we should keep `decode_geom` as a parameter or move into a different function.

Reply:
One of the problems with having a separate method would be that eventually you had to iterate the whole dataframe twice: one in `postprocess_dataframe` and eventually one for `decode_geom`.

Reply:
If I am not wrong, we are already doing this way

Reply:
mmm you are right. I'd leave it as it is, so we can keep the same behaviour as in previous versions, once we decide the `new API`:tm: we'd probably make some changes like this.

Reply:
Yeah I don't know the best solution here. The goal of this is to make it easy to get a GeoDataFrame if a user wants without having geopandas as a dependency (since it's a heavy/complex install).

Shapely/geopandas are the PostGIS of the Python world so it's important to have that easy parity between CARTO and cartoframes/Python environment.

Currently a user can get a GeoDataFrame with this:
```python
import geopandas as gpd
gdf = gpd.GeoDataFrame(cc.read('mytable', decode_geom=True))
```

I wonder if we have a method to `Dataset` like `.to_gdf` (or something similar) it would make it more straightforward / easier in the code.

I also think that `decode_geom` is not an obvious operation.  Anyway, definitely tons of room for improvement here.

Reply:
The `to_gdf` method could make sense, but we should add geopandas as a dependency of the library :) (not sure if we want to do that, probably it's not a bad idea).

Anyway, I'd add all this improvements to the new API doc.

Reply:
Adding to API doc sounds good.

FWIW, pandas has several `to_x` methods (e.g., [df.to_feather](https://github.com/pandas-dev/pandas/blob/9821b77de692716d7c2b62db1a68cac9ffc456c3/pandas/io/feather_format.py#L50)) that only work if the user already has a package installed that's not an install requirement for the library. The idea is to offer optional functionality without offering a bloated base install. I really like this approach.